### DISCUSSION TITLE
[ui] combobox: clearing the selected item manually does not instantly clear the search input

### DISCUSSION THREAD

**Discussion by gdebrauwer** [alpinejs/alpine#3724] [7. Bugs (archived)] [answered]
When you manually clear the selected item (and even the search query), the search input is not instantly cleared. It still shows the old search value, and only removes it when click outside the combobox.

**Comment by cbaconnier** [comment]
I'm experimenting the same issue when changing the `selected` value with Livewire `@entangle`. 
Reference: https://github.com/livewire/livewire/discussions/6606#discussioncomment-7001676

But this is definitely an issue with Alpine - Combobox.

**Comment by SimoTod** [comment]
The search input is just a text fiels. It's not supposed to autoclear based
on other values unless you manually implement the logic for that. Can you
guys describe in more detail what you are trying to achieve?

**Reply by cbaconnier** [reply]
I my case:

1. Clearing the selected value with a button.
2. Updating the selected value from livewire.


I suspect both are the same issue and both can be observed in the video I shared https://github.com/livewire/livewire/discussions/6606#discussioncomment-7001676

--- 
**E.g. of usage (without livewire)**  
Clearing the selected value with a button.

- I have a combobox with the attribute `nullable`
- The only current way to clear the selected item it is to delete by pressing backspace. 
- I would like to add a button in the combobox to clear the selected item.
- But when I clear the selected item with `selected = null`, the selected item is still displayed the in the input until I click somewhere in the page.

To reproduce: _Using the example of the  [demo page](https://alpinejs.dev/component/headless-combobox), I've only added this button next to `x-combobox:button` and added the `nullable` attribute._
```blade
<button class="absolute inset-y-0 right-0 flex items-center mr-10" x-on:click="selected = null;">x</button>
```
--- 
**E.g. of usage (with livewire)** 
Updating the selected value from livewire.

- I want to filter some projects by selecting a user in the combobox. 
- In order to achieve that, I use `entangle` on the `selected` value of the combobox to update the variable on Livewire, filter the projects and update the query string. This works.
- But then if I hit refresh, the variable on Livewire will successfully get the value from the query string and returning it back to the combobox, but the input combobox will not display the selected item, until I click somewhere in the page.

It will happen the same if I change the value from Livewire.

**Reply by SimoTod** [reply]
(non-livewire) 
yeah, but what is displayed in the input doesn't have a direct correlation with what is selected. For instance, you can type anything in your input and it doesn't affect the selected value until you actually click on the item to select, the input reset to the selected values only on certain specific events to prevent the value from changing while the user is actually trying to type something different. 

If you want to add an x that clears the input and reset the selection, you need to do both actions in your button callback. I believe you can do `x-on:click="selected = null; __resetInput();"`

**Reply by cbaconnier** [reply]
`__resetInput()` doesn't seems to do the job because `x-on:click="selected = null; console.log(__getCurrentValue())"` (on which is based resetInput) still returns the previous value.
But `x-on:click="selected = null; $refs.__input.value = ''"` will work

**Comment by gdebrauwer** [comment, marked as answer]
I created a PR that fixes this issue: https://github.com/alpinejs/alpine/pull/3934
